tests/test_h2_decoder.c

```diff
@@ -293,6 +293,61 @@ H2_DECODER_ON_CLIENT_TEST(h2_decoder_data_ignores_unknown_flags) {
     return AWS_OP_SUCCESS;
 }
 
+H2_DECODER_ON_CLIENT_TEST(h2_decoder_data_payload_max_size_update) {
+    (void)allocator;
+    struct fixture *fixture = ctx;
+    /* the initial max size is set as 16384, let's create a data frame with 16500 bytes data */
+    aws_h2_decoder_set_setting_max_frame_size(fixture->decode.decoder, 16500);
+    /* clang-format off */
+    uint8_t input[16509] = {
+        0x00, 0x40, 0x74,           /* Length (24) */
+        AWS_H2_FRAME_T_DATA,        /* Type (8) */
+        AWS_H2_FRAME_F_END_STREAM,  /* Flags (8) */
+        0x76, 0x54, 0x32, 0x10,     /* Reserved (1) | Stream Identifier (31) */
+        /* DATA */
+    };
+    /* clang-format on */
+    /* set the data and expected to 16500 'a' */
+    char expected[16501];
```

Comment:
trivial: it's weird to be testing exactly 16500 bytes with a 166501 byte buffer.
we don't need to encode/decode an EOS.
This function `h2_decode_tester_check_data_across_frames()` works without the data being an actual null-terminated c-str